server/tests/db/test_connection_pool.py

```
@@ -230,16 +239,17 @@ def test_failed_reset_does_not_block_putconn(self, fresh_pool, flask_app):
         pool, factory = fresh_pool
         connection, cursor = _make_conn()
         factory.return_value.getconn.return_value = connection
-        cursor.execute.side_effect = [None, None, Exception("conn lost")]
+        # SET ROLE succeeds, SET user_id succeeds, SET org_id succeeds, RESET raises
+        cursor.execute.side_effect = [None, None, None, Exception("conn lost")]
 
         with flask_app.test_request_context(
             "/api/x", headers={"X-User-ID": "u-1", "X-Org-ID": "org-7"},
         ):
             with pool.get_connection() as conn:
```

Ruff (0.15.20)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)